libvast/vast/detail/concepts.hpp

```diff
@@ -41,4 +56,31 @@ concept unsigned_integral = integral<T> && std::is_unsigned_v<T>;
 template <class T>
 concept signed_integral = integral<T> && std::is_signed_v<T>;
 
+struct any_callable {
+  template <class... Ts>
+  void operator()(Ts&&...);
+};
+
+/// Inspectables
+template <class T>
+concept is_inspectable = requires(detail::any_callable& i, T& x) {
+  {inspect(i, x)};
+};
+
+/// push_back
+template <class C>
+concept has_insert = requires(C xs, typename C::value_type x) {
+  xs.insert(x);
+};
+
+template <class C>
+concept has_push_back = requires(C xs, typename C::value_type x) {
```

Comment:
I'm not sure this or `has_insert` are great concepts. Code-wise, they're correct and fine, but I may personally prefer a simple in-situ requires expression on the function template using this.

Reply:
I propose to rename `has_insert` to `insertable` and `has_push_back` to `appendable`.

Reply:
Implemented as proposed.